ipatests/test_integration/test_ipa_ipa_migration.py

```diff
@@ -922,20 +869,195 @@ def test_ipa_migrate_stage_mode_with_invalid_cert(self):
             self.master.config.admin_password,
-            extra_args=['-n']
+            extra_args=['-n'],
+        )
+        install_msg = self.replicas[0].get_file_contents(
+            paths.IPA_MIGRATE_LOG, encoding="utf-8"
+        )
+        assert result.returncode == 0
+        assert MIGRATION_SCHEMA_LOG_MSG in install_msg
+        assert MIGRATION_DATABASE_LOG_MSG in install_msg
+        assert IPA_UPGRADE_LOG_MSG in install_msg
+        assert SIDGEN_TASK_LOG_MSG in install_msg
+
+    def test_ipa_migrate_prod_mode_hbac_rule(self):
+        """
+        This testcase checks that hbac rule is migrated from
+        remote server to local server in prod mode.
+        """
+        hbac_rule_name1 = 'test1'
+        hbac_rule_name2 = 'testuser_sshd'
+        tasks.kinit_admin(self.replicas[0])
+        cmd1 = self.replicas[0].run_command(
+            ["ipa", "hbacrule-find", hbac_rule_name1])
+        cmd2 = self.replicas[0].run_command(
+            ["ipa", "hbacrule-find", hbac_rule_name2])
+        assert hbac_rule_name1 in cmd1.stdout_text
+        assert hbac_rule_name2 in cmd2.stdout_text
+
+    def test_ipa_migrate_prod_mode_sudo_rule(self):
+        """
+        This testcase checks that sudo cmd and rules are
+        migrated from remote server to local server in prod mode.
+        """
+        sudorule = 'readfiles'
+        sudocmd = '/usr/bin/less'
+        tasks.kinit_admin(self.replicas[0])
+        cmd1 = self.replicas[0].run_command(
+            ["ipa", "sudorule-find", sudorule])
+        cmd2 = self.replicas[0].run_command(
+            ["ipa", "sudocmd-find", sudocmd])
+        assert 'Rule name: readfiles\n' in cmd1.stdout_text
+        assert 'Sudo Command: /usr/bin/less\n' in cmd2.stdout_text
+
+    def test_ipa_migrate_prod_mode_new_user_sid(self):
+        """
+        This testcase checks that in prod-mode uid/gid of the
+        migrated user is preserved i.e we have same
+        uid/gid for user on remote and local IPA server.
+        """
+        username = 'testuser4'
+        tasks.clear_sssd_cache(self.master)
+        result1 = self.master.run_command(['id', username])
+        tasks.clear_sssd_cache(self.replicas[0])
+        result2 = self.replicas[0].run_command(['id', username])
+        assert result1.stdout_text == result2.stdout_text
+
+    def test_check_vault_is_migrated(self):
+        """
+        This testcase checks that vault is
+        not migrated
+        """
+        vault_name = "testvault"
+        CMD_OUTPUT = (
+            "----------------\n"
+            "0 vaults matched"
+            "----------------\n"
+            "----------------------------\n"
+            "Number of entries returned 0\n"
+            "----------------------------"
         )
+        cmd = self.replicas[0].run_command(
+            ["ipa", "vault-find", vault_name], raiseonerr=False)
+        assert cmd.returncode != 0
+        assert CMD_OUTPUT not in cmd.stdout_text
```

Comment:
The test is passing because there is an error in CMD_OUPUT (a `\n` is missing at the end of the line `0 vaults matched`.
Replace with `CMD_OUTPUT="0 vaults matched"` and `assert CMD_OUTPUT in cmd.stdout_text`

Reply:
Replace `assert CMD_OUTPUT not in cmd.stdout_text` with `assert CMD_OUTPUT in cmd.stdout_text` (because ipa-migrate should not migrate the vault entries).